src/openforms/forms/tests/variables/test_variables_in_logic_trigger.py

```diff
@@ -117,6 +116,46 @@ def test_invalid_user_defined_variable_in_trigger(self):
             error["invalidParams"][0]["reason"],
         )
 
+    def test_invalid_date_format_for_user_defined_variable(self):
+        user = SuperUserFactory.create()
+        form = FormFactory.create()
+        FormVariableFactory.create(
+            source=FormVariableSources.user_defined,
+            key="date_var",
+            form=form,
+            data_type=FormVariableDataTypes.date,
+        )
+        form_logic_data = [
+            {
+                "form": f"http://testserver{reverse('api:form-detail', kwargs={'uuid_or_slug': form.uuid})}",
+                "order": 0,
+                "json_logic_trigger": {"!!": [True]},
+                "actions": [
+                    {
+                        "variable": "date_var",
+                        "action": {
+                            "type": "variable",
+                            "value": "07-03-2023",  # not isoformat
+                        },
+                    }
+                ],
+            }
+        ]
+
+        self.client.force_authenticate(user=user)
+        url = reverse("api:form-logic-rules", kwargs={"uuid_or_slug": form.uuid})
+        response = self.client.put(url, data=form_logic_data)
+
+        self.assertEqual(status.HTTP_400_BAD_REQUEST, response.status_code)
+
+        error = response.json()
+
+        self.assertEqual("invalid", error["code"])
+        self.assertEqual(
+            "Value for date variable must be in the format yyyy-mm-dd",
+            error["invalidParams"][0]["reason"],
+        )
```

Comment:
I'd like to see an assertion on the param `name` here too, to ensure we have a valid data path for the frontend to render.